### PR TITLE
OAK-10332 Allow reusing FilterProviderImpl outside OSGi context

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

**Review by mreutegg — COMMENTED** [review]
Looks good to me, but would be good to also have feedback from @anchela. She knows this code better than I do.

File: oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java [hunk 1]
```diff
@@ -76,6 +76,15 @@ public class FilterProviderImpl implements FilterProvider {
     private final Map<String, String> validatedPrincipalNamesPathMap = Maps.newConcurrentMap();
     private final Map<String, String> unsupportedPrincipalNames = Maps.newConcurrentMap();
 
+    /**
+     * Constructor to use outside OSGi containers
+     * @param oakPath the repository path where the principals are located
+     * @since 1.54
+     */
+    public FilterProviderImpl(String oakPath) {
```
Comment (anchela) [inline oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java]: @kwin , please add @NotNull annotation for the path param

File: oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java [hunk 2]
```diff
@@ -98,19 +107,23 @@ public Filter getFilter(@NotNull SecurityProvider securityProvider, @NotNull Roo
...
     //----------------------------------------------------< SCR Integration >---
 
+    public FilterProviderImpl() {
+        // constructor to use from SCR (not yet possible to use constructor injection, see https://issues.apache.org/jira/browse/OAK-9837)
+    }
+
     @Activate
-    protected void activate(Configuration configuration, Map<String, Object> properties) {
-        setPath(configuration);
+    protected void activate(Configuration configuration) {
```
Comment (anchela) [inline oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java]: removing the unused property map is an unrelated change, right? i would not do it as part of this PR but create a separate task for that.

File: oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java [hunk 3]
```diff
@@ -98,19 +107,23 @@ public Filter getFilter(@NotNull SecurityProvider securityProvider, @NotNull Roo
...
+        setPath(configuration.path());
     }
 
     @Modified
-    protected void modified(Configuration configuration, Map<String, Object> properties) {
-        setPath(configuration);
+    protected void modified(Configuration configuration) {
```
Comment (anchela) [inline oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java]: same as above

File: oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java [hunk 4]
```diff
@@ -98,19 +107,23 @@ public Filter getFilter(@NotNull SecurityProvider securityProvider, @NotNull Roo
...
+        setPath(configuration.path());
     }
 
-    private void setPath(@NotNull Configuration configuration) {
-        checkState(isValidPath(configuration.path()), "Configured path must be a valid absolute path.");
-        oakPath = configuration.path();
+    private void setPath(String path) {
```
Comment (anchela) [inline oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java]: please add @Nullable or @NotNull annotation. was there a reason for removing the annotation?

**Review by anchela — CHANGES_REQUESTED** [review]

File: oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java (hunk 1 above)
Comment (kwin) [reply oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java]: Done

File: oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java (hunk 2 above)
Comment (kwin) [reply oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java]: Sorry, I will revert if you insist but will not do the additional overhead of creating another PR. For me things like these should be fixed along with the issues. In this case I fail to see risk of regression.

File: oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java (hunk 4 above)
Comment (kwin) [reply oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java]: No reason, I restored the nullness annotation now.

**Review by anchela — APPROVED** [review]
lgtm
